rewrite-core/src/main/java/org/openrewrite/marker/RecipesThatMadeChanges.java

```diff
@@ -32,4 +34,70 @@ public class RecipesThatMadeChanges implements Marker {
     UUID id;
 
     Collection<Stack<Recipe>> recipes;
+
+    /**
+     * A list of recipe names and their associated options which have made changes to the AST
+     * @param indent string prepended to sub recipes for logging a hierarchical view
+     */
+    @Incubating(since = "7.22.0")
+    public List<String> recipeNamesAndOptionsThatMadeChanges(String indent) {
+        List<RecipeDescriptor> recipesToDisplay = new ArrayList<>();
+
+        for (Stack<Recipe> currentStack : recipes) {
+
+            Recipe root = currentStack.get(1);
+            RecipeDescriptor rootDescriptor = root.getDescriptor().withRecipeList(new ArrayList<>());
+
+            RecipeDescriptor index;
+            if (!recipesToDisplay.contains(rootDescriptor)) {
+                recipesToDisplay.add(rootDescriptor);
+                index = rootDescriptor;
+            } else {
+                index = recipesToDisplay.get(recipesToDisplay.indexOf(rootDescriptor));
+            }
+
+            for (int i = 1; i < currentStack.size(); i++) {
+                Recipe child = currentStack.get(i);
+                if (child == root) {
+                    continue;
+                }
+                RecipeDescriptor childDescriptor = child.getDescriptor().withRecipeList(new ArrayList<>());
+                if (!index.getRecipeList().contains(childDescriptor)) {
+                    index.getRecipeList().add(childDescriptor);
+                    index = childDescriptor;
+                } else {
+                    index = index.getRecipeList().get(index.getRecipeList().indexOf(childDescriptor));
+                }
+            }
+        }
+        String prefix = "" + indent;
```

Comment:
I would just move the code that converts the recipe descriptors to a string representation to the maven and Gradle plugins, there is really not much code here. The model should not really dictate how to render the descriptors.